behat/tests/bootstrap/v2/common/ElementSelectionTrait.php

```diff
@@ -94,4 +95,32 @@ public function iSelectBasedOnChoiceNumber(string $elementType, string $attribut
 
         return $choices[$choiceNumber];
     }
+
+    // Select radio dialogue based on name
+    public function iSelectRadioBasedOnName(string $elementType, string $attributeType, string $attributeValue, string $name)
+    {
+        $xpath = sprintf("//%s[@%s='%s']", $elementType, $attributeType, $attributeValue);
+        $session = $this->getSession();
+        $element = $session->getPage()->find(
+            'xpath',
+            $session->getSelectorsHandler()->selectorToXpath('xpath', $xpath)
+        );
+
+        if (null === $element) {
+            throw new \InvalidArgumentException(sprintf('Could not evaluate XPath: "%s"', $xpath));
+        }
+
+        $values = $element->findAll('css', 'input');
```

Comment:
Why not jus modify your initial xpath to have a trailing `//input`?

Reply:
Yeah I seem to have started writing it to be fully re-usable then scuppered it and made it not as re-useable :) Will change.